src/BSON/Timestamp.c

```diff
@@ -46,6 +46,36 @@ PHONGO_API zend_class_entry *php_phongo_timestamp_ce;
 
 zend_object_handlers php_phongo_handler_timestamp;
 
+/* Initialize the object from a string and return whether it was successful. */
+static bool php_phongo_timestamp_initialize(php_phongo_timestamp_t *intern, phongo_long increment, phongo_long timestamp)
```

Comment:
The range check to ensure these numbers are 4-byte integers (i.e. `(x < 0 || x > UINT32_MAX)`) only takes place in the constructor. I think a test should be added to ensure we cannot pass larger values through `unserialize()` and `__set_state()` on 64-bit, even though they will be truncated in the assignments below.